src/contexts/Cart.tsx

```diff
@@ -20,16 +23,63 @@ export const startingCart = {
       quantity: 1,
     },
   ],
-
-  setCart: (): void => undefined,
+  removeItem: (): void => undefined,
+  changeQuantity: (): void => undefined,
+  addItem: (): void => undefined,
 };
 
+export const removeItemFactory =
+  (setCart: React.Dispatch<React.SetStateAction<Cart>>) =>
+  (id: string): void => {
+    setCart((currCart) => currCart.filter((cartItem) => id !== cartItem.id));
+  };
+
+export const changeQuantityFactory =
+  (setCart: React.Dispatch<React.SetStateAction<Cart>>) =>
+  (id: string, amount: number): void => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          const newQty =
+            currItem.quantity + amount > 0 ? currItem.quantity + amount : 0;
+          return { ...currItem, quantity: newQty };
+        }
+        return { ...currItem };
+      });
+    });
+  };
+
+export const addItemFactory =
+  (cart: Cart, setCart: React.Dispatch<React.SetStateAction<Cart>>) =>
+  (product: Product, amount: number): void => {
+    const productInCart = cart.find((item) => item.id === product.id);
+    if (productInCart) {
```

Comment:
Do we need this check? When the product is already in the cart, we can use the change quantity function. When it's not, we can use addItem

Reply:
on it!